### ISSUE TITLE
proxy.R/session.R - pass-through to sessions

### ISSUE THREAD

**Issue by s-u** [att/rcloud#1678]
Support a way to get back to a session regardless of the cloud node it is on. Depends on #1677

**Comment by s-u** [comment]
What do we want to do here? We have to define what the session provides - the most reasonably implementable are HTTP requests that get passed through to something on the host running the session - but what exactly that has to be is yet to be defined. Any ideas?

**Comment by gordonwoodhull** [comment]
Yes, the most common things that break are Ajax calls and accessing of file assets. All HTTP. 

I am hoping that the UI library does not need to be configured in any way. It opens a port and the port gets forwarded automatically. This is because there are dozens of these R packages (and Python modules) which provide and it would be great to support ones we don't know about.

**Comment by s-u** [comment]
Ok, but somehow we need to mangle the URLs that the packages create to RCloud URLs so the question is how do we do that? How do they declare the URL and how can we change it? Let's say we have the `proxy.R` as above - is that all we need?

**Comment by gordonwoodhull** [comment]
If we start with that, it ought to handle packages that use relative URLs fine. 
I don't yet know if some packages use absolute URLs - it seems likely. 
I guess there may be no way to deal with that.

**Comment by s-u** [comment]
Ok, we now have `proxy.R` with

```
proxy.R/<user>/<session>[:<port>]/[<path>]  
```

forwarding to `http://<session-host>:<port>/<path>`

I have no idea how to test this in the context of shiny/htmlwidgets so please have a look and let  me know if that addresses the issue.

**Comment by s-u** [comment]
Technical note: depending on the setup, script-processing R instance may not have RCS loaded, so `proxy.R` is using `session.init.rcs()`. It actually works, because scripts need config access anyway to run `configure.rcloud` (even if they are proxified), but it would be more clean to have that handled by the script launching code.

**Comment by gordonwoodhull** [comment]
This feature works. I think the only reason the issue is still open is that we haven't successfully applied the feature to a new rcloud.shiny implementation. Closing it.
